apps/site/src/pages/account/api-keys-panel/types.ts

```diff
@@ -0,0 +1,34 @@
+import { Dispatch, ReactElement, SetStateAction } from "react";
+
+import { UserFacingApiKeyProperties } from "../../../lib/api/model/api-key.model";
+
+export type ApiKeyProps = UserFacingApiKeyProperties;
+
+export type KeyAction = "rename" | "revoke";
+
+export type KeyActionStatus =
+  | {
+      publicId: string;
+      action: KeyAction;
+    }
+  | undefined;
+
+export interface ApiKeyItemProps {
+  renameApiKeyCard: ReactElement;
+  revokeApiKeyCard: ReactElement;
+  matchingNewlyCreatedKey?: string;
+  apiKey: ApiKeyProps;
+  keyAction?: KeyAction;
+}
+
+export interface ApiKeysContextValue {
+  apiKeys: ApiKeyProps[];
+  setApiKeys: Dispatch<SetStateAction<ApiKeyProps[]>>;
+  isCreatingNewKey: boolean;
+  setIsCreatingNewKey: Dispatch<SetStateAction<boolean>>;
+  newlyCreatedKeyIds: string[];
+  setNewlyCreatedKeyIds: Dispatch<SetStateAction<string[]>>;
+  keyActionStatus: KeyActionStatus;
```

Comment:
It's really unclear to me why this needs to be in context at all – can it not just be the state in the card itself?

Reply:
Refer to this comment https://github.com/blockprotocol/blockprotocol/pull/1257#discussion_r1169186761

Reply:
resolving here, related changes will be implemented at https://github.com/blockprotocol/blockprotocol/pull/1271